clang/lib/Format/ContinuationIndenter.cpp

```diff
@@ -803,6 +803,60 @@ void ContinuationIndenter::addTokenOnCurrentLine(LineState &State, bool DryRun,
     return !Tok.Previous->isOneOf(TT_CastRParen, tok::kw_for, tok::kw_while,
                                   tok::kw_switch);
   };
+  // Detecting functions is brittle. It would be better if we could annotate
+  // the LParen type of functions/calls.
```

Comment:
Did you try it?

Reply:
I took a look, but didn't see a straightforward way to add support inside of `parseParens`

Reply:
I've done it in #97938. Can you use it after it's merged?

Reply:
> I've done it in #97938. Can you use it after it's merged?

Yes, that works to replace the `IsFunctionDeclParen`, very nice thanks. I still need the hacky way of detection FunctionCallLParen.